src/utils/settingsSync.ts

```diff
@@ -121,3 +126,169 @@ export async function uploadSettingsBackup(showToast = true): Promise<void> {
         setImmediate(() => document.body.removeChild(input));
     }
 }
+
+// Cloud settings
+const cloudSettingsLogger = new Logger("Cloud:Settings", "#39b7e0");
+
+export async function putCloudSettings() {
+    const settings = await exportSettings();
+
+    try {
+        const res = await fetch(new URL("/v1/settings", getCloudUrl()), {
+            method: "PUT",
+            headers: new Headers({
+                Authorization: await getCloudAuth(),
+                "Content-Type": "application/octet-stream"
+            }),
+            body: deflateSync(strToU8(settings))
+        });
+
+        if (!res.ok) {
+            cloudSettingsLogger.error(`Failed to sync up, API returned ${res.status}`);
+            showNotification({
+                title: "Cloud Settings",
+                body: `Could not synchronize settings (API returned ${res.status}).`,
+                color: "var(--red-360)"
+            });
+            return;
+        }
+
+        const { written } = await res.json();
+        PlainSettings.cloud.settingsSyncVersion = written;
+        VencordNative.ipc.invoke(IpcEvents.SET_SETTINGS, JSON.stringify(PlainSettings, null, 4));
+
+        cloudSettingsLogger.info("Settings uploaded to cloud successfully");
+        showNotification({
+            title: "Cloud Settings",
+            body: "Synchronized your settings!",
+            color: "var(--green-360)"
+        });
+    } catch (e: any) {
+        cloudSettingsLogger.error("Failed to sync up", e);
+        showNotification({
+            title: "Cloud Settings",
+            body: `Could not synchronize settings (${e.toString()}).`,
+            color: "var(--red-360)"
+        });
+    }
+}
+
+export async function getCloudSettings(shouldNotify = true, force = false) {
+    try {
+        const res = await fetch(new URL("/v1/settings", getCloudUrl()), {
+            method: "GET",
+            headers: new Headers({
+                Authorization: await getCloudAuth(),
+                Accept: "application/octet-stream",
+                "if-none-match": Settings.cloud.settingsSyncVersion.toString()
```

Comment:
soooo inconsistent :(
```suggestion
                "If-None-Match": Settings.cloud.settingsSyncVersion.toString()
```